Audit fixes final

ValliC444 opened the pull request:
Made changes to fix some of the issues raised during audit - disabling autocomplete for login, changing delete user api to post, edit user access only for the creator.

Commit:
Edit user html page was displaying even the users the current user was not authorised to edit(only displaying, the user was still unable to update because the update function had checks to restrict out of scope).
Fixed it by adding a check for user access level in backend, and redirecting to user management page in frontend if the response from backend is null.

Commit:
autocomplete off for login form

Commit:
changed delete user api to post, added token

Commit:
minor changes

Commit:
added comments wherever applicable

anuranjan-alation commented on Email/NMSReportingSuite/pom.xml:
```diff
@@ -143,34 +143,35 @@
             <groupId>org.apache.shiro</groupId>
             <artifactId>shiro-core</artifactId>
-            <version>LATEST</version>
+            <version>1.4.1</version>
```
Why not chnge in the parameter rather than hard-coding here?

anuranjan-alation commented on NMSReportingSuite/pom.xml:
```diff
@@ -159,34 +159,35 @@ xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/ma
         <dependency>
             <groupId>org.apache.shiro.samples</groupId>
             <artifactId>samples-spring-client</artifactId>
-            <version>LATEST</version>
```
ditto

anuranjan-alation commented on NMSReportingSuite/src/main/java/com/beehyv/nmsreporting/controller/UserController.java:
```diff
@@ -510,12 +513,13 @@ public String forgotPassword(@RequestBody ForgotPasswordDto forgotPasswordDto, H
+    public Map deleteExistingUser(HttpServletRequest request, @RequestBody Integer id) {
         User currentUser = userService.getCurrentUser();
-        if(currentUser != null){
+        String token = "dhty" + currentUser.getUserId().toString() + "alkihkf";
```
is 'dhty' and 'alkijkf' random by choice? or can it be more meaningful?

ValliC444 replied:
It's random by choice. They just wanted some token verification, so I am using a random string combined with userId

anuranjan-alation commented on app/views/login.html:
```diff
@@ -68,14 +68,15 @@
-					<input name="password0" type="password"  value="Dummy" style="display: none" />
-					<input type="password" class="form-control password-input input input-sm login-inputs" id="divID" data-ng-model="user.password"  name="password" placeholder="Password"	 autocomplete="new-password" value="Dummy">
-					<input name="password1" type="password"  value="Dbmnmummy" style="display: none" />
+<!--					<input name="password0" type="password"  value="Dummy" style="display: none" />-->
```
if not required, please delete the commented code. We can always retrieve these back from the history

Commit:
Changed the default pswd on user creation to a randomly generated string and set up email to send the same to user. Same for admin pswd reset.

penugonda committed:
Merge pull request #26 from motech-implementations/changing_default_pswd_to_random

Changed the default pswd on user creation to a randomly generated str…

jangidrakesh committed:
Implement SameSite=Lax to prevent csrf attack using interceptor for all request under nms/**

jangidrakesh committed:
Issue fixed during security audit
1)Implement SameSite=Lax for all request
2)Implement csrf token for all importent request
3)Update privilige level of accessing issue in all user operations
4)Restrict accessing Mis application untill new password is set by new user
5)Block accessing templates by loging in as normal user
6)removed the password from resetPassword api's response
7)Allowed only for admin user to access this url "http://192.168.200.112:8080/NMSReportingSuite/nms/user/roles"
8)Fix issue in API rate limiting
9)removed tomcat banner grabbing

jangidrakesh committed:
redirecting to change password page for admin authorizes pages

jangidrakesh committed:
change view folder name
